### DISCUSSION TITLE
Best practice to maintain / sync list of trusted owners over multiple repositories

### DISCUSSION THREAD

**Discussion by Sam13** [NuGet/Home#12131] [Q&A]
I have several repositories (60+) for which I want to enforce signed NuGet packages and trust only certain owners from NuGet.org.
Currently I have a _nuget.config_ in each repo which needs to be updated whenever the list of trusted owners changes.

What's best practice to keep those list of owners in sync over all repos?

**Comment by erdembayar** [comment]
~~@joelverhagen 
Looks like this nuget.org question.~~
Are those trusted owners same or not for all 60 repositories?
If same why don't you have 1 nuget.config with trusted signers section on top folder of your repositories since nuget.config  file in an each repo hierarchly inherit from above one. 
If not then most likely you have to keep them separate for each repo.

**Reply by Sam13** [reply]
> Are those trusted owners same or not for all 60 repositories?

Yes, can be the same - mainly it contains the trusted owners for my organization.


> If same why don't you have 1 nuget.config with trusted signers section on top folder of your repositories since nuget.config file in an each repo hierarchly inherit from above one. 

Those repos are used in different places such as CI environments, local developer workstations.
There is no guarantee that a _nuget.config_ in the top folder exists in all that places - and if so, how do you sync that to all involved parties?

**Reply by erdembayar** [reply]
If repositories are scattered, then I imagine it would be hard. 
Some point your code going to CI/CD right? Then maybe add custom task verify if nuget.config in repo contains correct set of owners in it right before build the project. Since nuget.config file is just xml so you can parse with powershell or C#.
